h/groups/views.py

```diff
@@ -101,26 +104,28 @@ def _read_group(request, group):
     """
     url = request.route_url('group_read', pubid=group.pubid, slug=group.slug)
 
-    result = search.search(request,
-                           private=False,
-                           params={"group": group.pubid, "limit": 1000})
-    annotations = [presenters.AnnotationHTMLPresenter(
-                       storage.annotation_from_dict(a))
-                   for a in result['rows']]
-
-    # Group the annotations by URI.
-    # Create a dict mapping the (normalized) URIs of the annotated documents
-    # to the most recent annotation of each document.
-    annotations_by_uri = collections.OrderedDict()
-    for annotation in annotations:
-        normalized_uri = uri.normalize(annotation.uri)
-        if normalized_uri not in annotations_by_uri:
-            annotations_by_uri[normalized_uri] = annotation
-            if len(annotations_by_uri) >= 25:
-                break
-
-    document_links = [annotation.document_link
-                      for annotation in annotations_by_uri.values()]
+    # Find documents that have at least one annotation that:
+    # a. is shared, and
+    # b. is in this group.
+    documents = (
+        request.db.query(models.Document)
+        .options(joinedload(models.Document.meta))
+        .options(joinedload(models.Document.document_uris))
+        .filter(
+            models.Document.annotations.any(
+                and_(
+                    models.Annotation.shared == expression.true(),
```

Comment:
`== True` produces linter warnings from both Flake8 and PyLint. You're supposed to do `is True`. But `is True` doesn't work in this case (it never finds any annotations). Of course you could argue that the linters are wrong in this case. But the choice was between 1. Ignoring the linter warnings/creating linter noise, 2. Putting in linter comments to silence this line, or 3. Using `true()`.